### PR TITLE
docs: update/cleanup `Invite` documentation

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Fixes backticks, adds missing `Optional[]` types, and adds links back to the table explaining the conditions under which attributes exist.

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/invite.py [hunk 1]
```diff
@@ -275,32 +275,34 @@ class Invite(Hashable):
...
+    +------------------------------------+---------------------------------------------------------------------+
+    | :attr:`expires_at`                 | :meth:`Client.fetch_invite` with ``with_expiration`` enabled        |
+    +------------------------------------+---------------------------------------------------------------------+
+    | :attr:`guild_scheduled_event`      | :meth:`Client.fetch_invite` with valid ``guild_scheduled_event_id`` |
+    |                                    | or valid event ID in URL or invite object                           |
+    +------------------------------------+---------------------------------------------------------------------+
+
+    If something is not in the table above, then it's available by all methods.
 
     Attributes
```
Comment (Victorsitou) [inline disnake/invite.py:307]: `max_age` is missing `Optional[]` type below

Comment (shiftinv) [reply disnake/invite.py:307]: good catch, ty! https://github.com/DisnakeDev/disnake/pull/864/commits/9607d8af10a70b636e8f39344507a102570e832d

File: disnake/invite.py [hunk 2]
```diff
@@ -275,65 +275,89 @@ class Invite(Hashable):
...
     ----------
-    max_age: :class:`int`
-        How long before the invite expires in seconds.
-        A value of ``0`` indicates that it doesn't expire.
     code: :class:`str`
         The URL fragment used for the invite.
     guild: Optional[Union[:class:`Guild`, :class:`Object`, :class:`PartialInviteGuild`]]
         The guild the invite is for. Can be ``None`` if it's from a group direct message.
-    created_at: :class:`datetime.datetime`
+
```
Comment (Victorsitou) [inline disnake/invite.py:313]: these spaces seems unnecessary imo

Comment (shiftinv) [reply disnake/invite.py]: fair enough - https://github.com/DisnakeDev/disnake/pull/864/commits/d7a0bdbd12b66604dc461fad87e988a36ee04f02

**Review by Victorsitou — APPROVED** [review]

**Pull request merged**